dockerfiles/theia/Dockerfile

```diff
@@ -11,54 +11,62 @@
 ###
 # Builder Image
 #
-FROM node:8.12.0-alpine as builder
-
-ARG GITHUB_TOKEN
-
-# Install basic software used for checking github API rate limit
-RUN apk add --update --no-cache curl jq expect
+FROM ${BUILD_ORGANIZATION}/${BUILD_PREFIX}-theia-dev:${BUILD_TAG} as builder
+WORKDIR ${HOME}
 
 # define in env variable GITHUB_TOKEN only if it is defined
 # else check if github rate limit is enough, else will abort requiring to set GITHUB_TOKEN value
-ADD src/builder/check-github-limit.sh /tmp/builder/scripts/check-github-limit.sh
-RUN /tmp/builder/scripts/check-github-limit.sh
-
-# Install tools required to build theia
-RUN apk add make gcc g++ python git bash
+ARG GITHUB_TOKEN
 
 # Define upstream version of theia to use
 ARG THEIA_VERSION=0.3.17
 
-ENV THEIA_VERSION=${THEIA_VERSION} \
-    HOME=/home/theia
-WORKDIR ${HOME}
-
-# add eclipse che theia generator
-RUN yarn global add @eclipse-che/theia-generator@0.0.1-1543568125
-
+# Invalidate cache if there is a new commit
+ADD https://api.github.com/repos/theia-ide/theia/git/refs/tags/v${THEIA_VERSION} /tmp/.sha1-repository
```

Comment:
Maybe we should invalidate cache after checking github limit?

Reply:
the thing is that if we invalidate cache after checking limit, it means that we won't check limit on a second run (layers cached) so it'll probably fail when compiling. I don't know the best option